### PR TITLE
Add additional tests for ExecutionManager

### PR DESCRIPTION
**Pull request by 3vilhamster**
**What changed?**
Adding more tests to ExecutionManager.
Now covering GetExecutionInfo and UpdateExecution.
That includes a lot of deserialization and data conversion. I've added some sane checks, if we'll spot any missing fields, it should be easy enough extend the test with extra fields.

**Why?**
Improving code coverage.

**How did you test it?**
unit tests

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/persistence/execution_manager_test.go [hunk 1]
```diff
@@ -24,8 +24,13 @@ package persistence
 
 import (
 	"context"
+	"fmt"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/checksum"
+	"github.com/uber/cadence/common/types"
```
Comment (taylanisikdemir) [inline common/persistence/execution_manager_test.go:30]: make fmt

File: common/persistence/execution_manager_test.go [hunk 2]
```diff
@@ -263,3 +268,277 @@ func TestGetReplicationTasks(t *testing.T) {
 		})
 	}
 }
+
```
Comment (taylanisikdemir) [inline common/persistence/execution_manager_test.go:271]: tests have failed on CI https://github.com/uber/cadence/actions/runs/8382150830/job/22955320462?pr=5809

File: common/persistence/execution_manager_test.go (hunk 1 above)
Comment (3vilhamster) [reply common/persistence/execution_manager_test.go]: ohwell. I've tried adding gci as file watcher, but it makes code writing very weird. I'll try to figure out how to set a helper for myself. Or maybe a recommit hook.

File: common/persistence/execution_manager_test.go (hunk 2 above)
Comment (3vilhamster) [reply common/persistence/execution_manager_test.go:271]: Good catch.
The issue was caused by not setting UTC in time.Unix. Fixed.

**Review by taylanisikdemir — APPROVED** [review]

**Pull request merged**